make of Basic_rBoot fails in clean Sming if compiled first time with ENABLE_SSL=1

Hello devs,

after i cloned the new Sming 3.0.1 to a new folder and set the paths in Makefile-user.mk of Basic_rBoot example, make fails with error in linker...

terminal output:

none@ubuntu:/opt/Sming/samples/Basic_rBoot$ make
OC out/build/libmain2.a
make -C /opt/Sming/Sming/third-party/rboot
make[1]: Verzeichnis „/opt/Sming/Sming/third-party/rboot“ wird betreten
CC rboot-stage2a.c
LD /opt/Sming/samples/Basic_rBoot/out/build/rboot-stage2a.elf
E2 /opt/Sming/samples/Basic_rBoot/out/build/rboot-hex2a.h
CC rboot.c
LD /opt/Sming/samples/Basic_rBoot/out/build/rboot.elf
E2 /opt/Sming/samples/Basic_rBoot/out/firmware/rboot.bin
make[1]: Verzeichnis „/opt/Sming/Sming/third-party/rboot“ wird verlassen
C+ app/application.cpp
CC /opt/Sming/Sming/third-party/rboot/appcode/rboot-bigflash.c
CC /opt/Sming/Sming/third-party/rboot/appcode/rboot-api.c
AR out/build/app_app.a
LD out/build/app_0.out
/opt/esp-open-sdk/xtensa-lx106-elf/lib/gcc/xtensa-lx106-elf/4.8.5/../../../../xtensa-lx106-elf/bin/ld: cannot find -lsming
collect2: error: ld returned 1 exit status
/opt/Sming/Sming/Makefile-rboot.mk:380: die Regel für Ziel „out/build/app_0.out“ scheiterte
make: *** [out/build/app_0.out] Fehler 1


At line 380 in Makefile there are this lines:

$(TARGET_OUT_0): $(APP_AR)
	$(vecho) "LD $@"
	$(Q) $(LD) -L$(USER_LIBDIR) -L$(SDK_LIBDIR) -L$(BUILD_BASE) $(RBOOT_LD_0) $(LDFLAGS) -Wl,--start-group $(APP_AR) $(LIBS) -Wl,--end-group -o $@
	$(Q) $(STRIP) $@

I dont understand this! Sorry

--- a/.travis.yml
+++ b/.travis.yml
@@ -30,16 +30,15 @@ script:
   - export ESP_HOME=$TRAVIS_BUILD_DIR/opt/esp-alt-sdk  
   - export PATH=$PATH:$TRAVIS_BUILD_DIR/opt/esp-alt-sdk/utils/
   - cd $SMING_HOME
-  - make test
+  - make test 
   - cd $SMING_HOME/../
   - export SMING_HAS_CHANGED=`for i in $CHANGED_FILES; do  if [[ $i == Sming/* ]]; then echo 1; break; fi; done`
   - if [ ! -z "$SMING_HAS_CHANGED" ]; then CHANGED_PROJECTS=`cd $SMING_HOME/../samples; ls -d *`; fi
   - for i in $CHANGED_PROJECTS; do echo "Compiling $i"; make -C "samples/$i"; if [ $? -ne 0 ]; then exit 1; fi; done
   - cd $SMING_HOME
-  - make clean
+  - make clean samples-clean
   - make ENABLE_GDB=1
   - make Basic_Debug
-  - make clean
+  - make clean samples-clean
   - make ENABLE_CUSTOM_HEAP=1
-  - make -C $SMING_HOME/../samples/Basic_Blink clean
   - make Basic_Blink ENABLE_CUSTOM_HEAP=1

--- a/Sming/Makefile
+++ b/Sming/Makefile
@@ -338,7 +338,7 @@ else
 	$(Q) $(MAKE) --no-print-directory -C spiffy clean V=$(V)
 endif	
 
-test: all spiffy samples
+test: Basic_Blink Basic_rBoot Basic_Ssl Basic_HwPWM 
 
 docs/wiki/Home.md: 
 	$(vecho) "No Wiki submodule found. Using git to fetch it..."

--- a/Sming/Makefile-project.mk
+++ b/Sming/Makefile-project.mk
@@ -137,9 +137,11 @@ TARGET		= app
 
 THIRD_PARTY_DIR = $(SMING_HOME)/third-party
 
+SMING_FEATURES = none
 LIBSMING = sming
 ifeq ($(ENABLE_SSL),1)
 	LIBSMING = smingssl
+	SMING_FEATURES = SSL
 endif
 
 # which modules (subdirectories) of the project to include in compiling
@@ -185,7 +187,7 @@ ifeq ($(ENABLE_SSL),1)
 		AXTLS_FLAGS += -DSSL_DEBUG=1 -DDEBUG_TLS_MEM=1
 	endif
 	
-	CUSTOM_TARGETS += $(USER_LIBDIR)/lib$(LIBSMING).a include/ssl/private_key.h
+	CUSTOM_TARGETS += include/ssl/private_key.h
 	CFLAGS += $(AXTLS_FLAGS)  
 	CXXFLAGS += $(AXTLS_FLAGS)	
 endif
@@ -313,7 +315,7 @@ endef
 
 .PHONY: all checkdirs spiff_update spiff_clean clean
 
-all: checkdirs $(TARGET_OUT) $(SPIFF_BIN_OUT) $(FW_FILE_1) $(FW_FILE_2)
+all: $(USER_LIBDIR)/lib$(LIBSMING).a checkdirs $(TARGET_OUT) $(SPIFF_BIN_OUT) $(FW_FILE_1) $(FW_FILE_2)
 
 spiff_update: spiff_clean $(SPIFF_BIN_OUT)
 
@@ -352,7 +354,7 @@ $(APP_AR): $(OBJ)
 	$(Q) $(AR) cru $@ $^
 	
 $(USER_LIBDIR)/lib$(LIBSMING).a:
-	$(vecho) "Recompiling Sming with SSL support. This may take some time"
+	$(vecho) "(Re)compiling Sming. Enabled features: $(SMING_FEATURES). This may take some time"
 	$(Q) $(MAKE) -C $(SMING_HOME) clean V=$(V) ENABLE_SSL=$(ENABLE_SSL) SMING_HOME=$(SMING_HOME)
 	$(Q) $(MAKE) -C $(SMING_HOME) V=$(V) ENABLE_SSL=$(ENABLE_SSL) SMING_HOME=$(SMING_HOME)
 

--- a/Sming/Makefile-rboot.mk
+++ b/Sming/Makefile-rboot.mk
@@ -139,8 +139,10 @@ TARGET		= app
 THIRD_PARTY_DIR = $(SMING_HOME)/third-party
 
 LIBSMING = sming
+SMING_FEATURES = none
 ifeq ($(ENABLE_SSL),1)
 	LIBSMING = smingssl
+	SMING_FEATURES = SSL
 endif
 
 # which modules (subdirectories) of the project to include in compiling
@@ -205,7 +207,7 @@ ifeq ($(ENABLE_SSL),1)
 		AXTLS_FLAGS += -DSSL_DEBUG=1 -DDEBUG_TLS_MEM=1
 	endif
 	
-	CUSTOM_TARGETS += $(USER_LIBDIR)/lib$(LIBSMING).a include/ssl/private_key.h
+	CUSTOM_TARGETS += include/ssl/private_key.h
 	CFLAGS += $(AXTLS_FLAGS)  
 	CXXFLAGS += $(AXTLS_FLAGS)	
 endif
@@ -357,7 +359,7 @@ endef
 
 .PHONY: all checkdirs spiff_update spiff_clean clean
 
-all: checkdirs $(LIBMAIN_DST) $(RBOOT_BIN) $(RBOOT_ROM_0) $(RBOOT_ROM_1) $(SPIFF_BIN_OUT) $(FW_FILE_1) $(FW_FILE_2)
+all: $(USER_LIBDIR)/lib$(LIBSMING).a checkdirs $(LIBMAIN_DST) $(RBOOT_BIN) $(RBOOT_ROM_0) $(RBOOT_ROM_1) $(SPIFF_BIN_OUT) $(FW_FILE_1) $(FW_FILE_2) 
 
 $(RBOOT_BIN):
 	$(MAKE) -C $(THIRD_PARTY_DIR)/rboot
@@ -392,7 +394,7 @@ $(APP_AR): $(OBJ)
 	$(Q) $(AR) cru $@ $^
 
 $(USER_LIBDIR)/lib$(LIBSMING).a:
-	$(vecho) "Recompiling Sming with SSL support. This may take some time"
+	$(vecho) "(Re)compiling Sming. Enabled features: $(SMING_FEATURES). This may take some time"
 	$(Q) $(MAKE) -C $(SMING_HOME) clean V=$(V) ENABLE_SSL=$(ENABLE_SSL) SMING_HOME=$(SMING_HOME)
 	$(Q) $(MAKE) -C $(SMING_HOME) V=$(V) ENABLE_SSL=$(ENABLE_SSL) SMING_HOME=$(SMING_HOME)
 

